openshift/Dockerfile

```diff
@@ -19,32 +19,33 @@ ENV HOME=/opt/ansible \
     USER_NAME=ansible \
     USER_UID=1001
 
-RUN yum install -y \
-      ansible \
-      ansible-runner \
-      ansible-runner-http \
-      python3-kubernetes \
-      python3-openshift \
-      python3-oauthlib \
-      python3-requests-oauthlib \
-      python3-jmespath \
-      tini \
- && yum clean all \
- && rm -rf /var/cache/yum
+# cryptography packages cannot be installed in the OSBS environment using pip.
+# Thus installing it via rpm.
+RUN yum install -y python3-cryptography redhat-rpm-config openssl-devel \
+    cargo pkg-config rust libffi-devel python3-devel gcc python3-pip
 
-COPY --from=builder /go/src/github.com/openshift/ansible-operator-plugins/ansible-operator /usr/local/bin/ansible-operator
-COPY openshift/release/ansible/ansible_collections ${HOME}/.ansible/collections/ansible_collections
+# pip install will install a higher version of the requests module. However, pip fails
+# to uninstall the existing request module as it is installed from rpm. Thus, removing
+# the module before using pip install.
+RUN yum remove -y python3-requests
 
-COPY openshift/release/ansible/bin /usr/local/bin
+# Add steps for cachito
+ENV REMOTE_SOURCES=${REMOTE_SOURCES:-"./openshift/"}
+ENV REMOTE_SOURCES_DIR=${REMOTE_SOURCES_DIR:-"/cachito"}
+COPY ${REMOTE_SOURCES} ${REMOTE_SOURCES_DIR}
+COPY openshift/install-ansible.sh .
+RUN chmod +x /install-ansible.sh \
+    && ./install-ansible.sh
 
-RUN chmod +x /usr/local/bin/user_setup && \
-    ./usr/local/bin/user_setup
+COPY --from=builder /go/src/github.com/openshift/ansible-operator-plugins/ansible-operator /usr/local/bin/ansible-operator
+COPY openshift/release/ansible/ansible_collections ${HOME}/.ansible/collections/ansible_collections
 
 # Ensure directory permissions are properly set
-RUN mkdir -p ${HOME}/.ansible/tmp \
- && chown -R ${USER_UID}:0 ${HOME} \
- && chmod -R ug+rwx ${HOME}
+RUN echo "${USER_NAME}:x:${USER_UID}:0:${USER_NAME} user:${HOME}:/sbin/nologin" >> /etc/passwd \
+  && mkdir -p ${HOME}/.ansible/tmp \
+  && chown -R ${USER_UID}:0 ${HOME} \
+  && chmod -R ug+rwx ${HOME}
 
 WORKDIR ${HOME}
 USER ${USER_UID}
-ENTRYPOINT ["/usr/bin/tini", "--", "/usr/local/bin/ansible-operator", "run", "--watches-file=./watches.yaml"]
```

Comment:
These two comments summarize it well regarding when to use tini and when not to:
- https://github.com/krallin/tini/issues/8#issuecomment-146135930
- https://github.com/krallin/tini/issues/8#issuecomment-146136579

Reply:
Is the claim here that there are no possibilities of zombie processes or mis-handling of signals? If it is completely assessed and tested, tini may not be needed.

Reply:
I see the upstream image still uses tini. I think we should try to keep the `ENTRYPOINT` the same unless specifically required. 
https://github.com/openshift/ansible-operator-plugins/blob/c07954c49d6a40590341191fe16e1646701f9844/images/ansible-operator/Dockerfile#L88

Reply:
As the playbook commands are run using exec.command(), it is best to have safeguard in place for production environments to avoid any bugs in dependent libraries causing the issues.

/hold

Reply:
Using `catatonit` instead of `tini`. More context here: https://redhat-internal.slack.com/archives/CBBJY9GSX/p1726751465415449